component/module/tests/updates_test.go

```diff
@@ -0,0 +1,162 @@
+package tests
```

Comment:
I've created these tests, because I've noticed there was non-trivial logic in the `loader.go` for updating components' exports, but I didn't see tests.

Reply:
Tests for exports might be located somewhere else (due to some historical artifacts about the iteration of modules). Check with @erikbaranowski to see if we can remove/deduplicate some tests now that we're adding these.

Reply:
Thanks! I've discussed with Erik and we've moved this test to `pkg/flow/module_caching_test.go`, as it mainly focuses on verifying the module-related caching in `loader.go`, especially in presence of concurrent updates.

Reply:
we decided to move these to the same folder as other module tests. I think they have value in testing some additional things and should be kept.

- loading more than just a module config by loading a config plus module config together. `arguments` and `exports` are passed between them
- testing a real module loader `module.string`. this is more a side effect of the testing and maybe module loaders should have their own testing eventually in `component/module`. Either way, it doesn't hurt to test more code along the way.